File: src/android/com/phemium/sipvideocall/data/CallData.java
package com.phemium.sipvideocall.data;

import com.phemium.sipvideocall.Utils;
import com.phemium.sipvideocall.constant.Constant;
import org.json.JSONException;
import org.json.JSONObject;

/**
 * Created by Tom on 1/5/2017.
 */

public class CallData extends Object {
    public String username;
    public String password;
    public String domain;
    public String proxy;
    public String turnServer0; // Turn Server #0 configuration
    public String turnDomain0;
    public String turnUsername0;
    public String turnPassword0;
    public String turnServer1; // Turn Server #1 configuration
    public String turnDomain1;
    public String turnUsername1;
    public String turnPassword1;
    public String toAddress;

    public Boolean onlyAudio;
    public Integer downloadBandwidth;
    public Integer uploadBandwidth;
    public Integer framerate;

    public String language;
    public String mainColor;
    public String secondaryColor;
    public int fontSize;
    public String fontColor;
    public String consultantName;
    public String displayTopViewMode;
    public int displayButtonTime;
    public String callRecordingNotificationVisible;

    public String transportMode;
    public String encryptionMode;
    public String videoSize;
    public String chatMode = "";
    public String zoomMode = "";
    public String videoCallPluginVersion;
    public String enduserPluginVersion;
    public String consultationId;
    public boolean logEnable;
    public String toEmail;

    public String permission;

    public CallData() {
        downloadBandwidth = -1;
        uploadBandwidth = -1;
        framerate = -1;
    }

    public int setCredentialValues(JSONObject credentialValues) {
        try {
            username = credentialValues.getString("username");
            password = credentialValues.getString("password");
            domain = credentialValues.getString("domain");

            proxy = credentialValues.getString("proxy");
            turnServer0 = credentialValues.getString("turnServer0");
            turnDomain0 = credentialValues.getString("turnDomain0");
            turnUsername0 = credentialValues.getString("turnUsername0");
            turnPassword0 = credentialValues.getString("turnPassword0");
            turnServer1 = credentialValues.getString("turnServer1");
            turnDomain1 = credentialValues.getString("turnDomain1");
            turnUsername1 = credentialValues.getString("turnUsername1");
            turnPassword1 = credentialValues.getString("turnPassword1");
        } catch (JSONException e) {
            e.printStackTrace();
            return Constant.DATA_PARSING_ERROR;
        }
        return Constant.NO_ERROR;
    }

    public int setCallSettingValues(JSONObject settingValues) {
        try {
            transportMode = settingValues.getString("transport_mode");
        } catch (JSONException e) {
            transportMode = "tls";
        }
        try {
            encryptionMode = settingValues.getString("encryption_mode");
        } catch (JSONException e) {
            encryptionMode = "none";
        }
        try {
            videoSize = settingValues.getString("video_size");
        } catch (JSONException e) {
            videoSize = "vga";
        }
        try {
            onlyAudio = settingValues.getBoolean("only_audio");
            downloadBandwidth = settingValues.getInt("download_bandwidth");
            uploadBandwidth = settingValues.getInt("upload_bandwidth");
            framerate = settingValues.getInt("framerate");
        } catch (JSONException e) {
            e.printStackTrace();
            return Constant.DATA_PARSING_ERROR;
        }
        return Constant.NO_ERROR;
    }

    public int setGUISettingValues(JSONObject guiSettingValues) {
        try {
            language = guiSettingValues.getString("language");
            mainColor = guiSettingValues.getString("main_color");
            secondaryColor = guiSettingValues.getString("secondary_color");
            fontSize = guiSettingValues.getInt("font_size");
            fontColor = guiSettingValues.getString("font_color");
            consultantName = guiSettingValues.getString("consultant_name");
            displayTopViewMode = guiSettingValues.getString("display_topview_mode");
            callRecordingNotificationVisible = guiSettingValues.getString("call_recording_notification_visible");
            chatMode = guiSettingValues.getString("chat_mode");
            zoomMode = guiSettingValues.getString("zoom_mode");
            logEnable =  guiSettingValues.getBoolean("log_mode");
        } catch (JSONException e) {
            e.printStackTrace();
        }
        this.checkGUIValues();
        return Constant.NO_ERROR;
    }

    public int setExtraSettingValues(JSONObject extraSettingValues) {

        try {
            toEmail = extraSettingValues.getString("extra_toemail");
            videoCallPluginVersion = extraSettingValues.getString("videocall_version");
            enduserPluginVersion = extraSettingValues.getString("enduser_version");
            consultationId = extraSettingValues.getString("consultation_id");
        } catch (JSONException e) {
        }
        return Constant.NO_ERROR;
    }
    public void setToAddress(String toAddressValue) {
        toAddress = toAddressValue;
    }

    public void checkGUIValues() {
        if (displayButtonTime <= 0) {
            displayButtonTime = 3;
        }
        if (Utils.isStringEmpty(mainColor)) {
            mainColor = "#0a79c7";
        }
        if (!mainColor.startsWith("#")) {
            mainColor = "#" + mainColor;
        }
        if (Utils.isStringEmpty(secondaryColor)) {
            secondaryColor = "#70c8e1";
        }
        if (!secondaryColor.startsWith("#")) {
            secondaryColor = "#" + secondaryColor;
        }
        if (Utils.isStringEmpty(displayTopViewMode)) {
            displayTopViewMode = "atScreenTouch";
        }
        if (Utils.isStringEmpty(fontColor)) {
            fontColor = "#ffffff";
        }
        if (!fontColor.startsWith("#")) {
            fontColor = "#" + fontColor;
        }
        if (fontSize <= 0) {
            fontSize = 17;
        }
        if (Utils.isStringEmpty(language)) {
            language = "es";
        }
        if (Utils.isStringEmpty(callRecordingNotificationVisible)) {
            callRecordingNotificationVisible = "no";
        }
        if (Utils.isStringEmpty(consultantName)) {
            consultantName = Constant.CONSULTANT_DEFAULT_NAME;
        }

        if (Utils.isStringEmpty(chatMode)) {
            chatMode = "WithChat";
        }
        if (Utils.isStringEmpty(zoomMode)) {
            zoomMode = "zoom";
        }
    }
}

File: src/ios/CallData.m
//
//  DataManager.m
//  SipVideoPluginTest
//
//  Created by SCNDev1 on 12/22/16.
//
//

#import "CallData.h"
#import "StringUtil.h"

@implementation CallData

@synthesize onlyAudioCall;
@synthesize download_bandwidth;
@synthesize upload_bandwidth;
@synthesize framerate;

//credentials
@synthesize username;
@synthesize password;
@synthesize domain;
@synthesize address;
@synthesize proxy;
@synthesize turnServer0;
@synthesize turnDomain0;
@synthesize turnUsername0;
@synthesize turnPassword0;
@synthesize turnServer1;
@synthesize turnDomain1;
@synthesize turnUsername1;
@synthesize turnPassword1;

//etras
@synthesize main_color;
@synthesize secondary_color;
@synthesize display_topview_mode;
@synthesize call_recording_notification_visible;
@synthesize display_button_time;
@synthesize language;
@synthesize consultant_name;
@synthesize font_size;
@synthesize font_color;

@synthesize transport_mode;
@synthesize encryption_mode;
@synthesize video_size;
@synthesize chat_mode;
@synthesize zoom_mode;
@synthesize videocall_version;
@synthesize enduser_version;
@synthesize consultation_id;
@synthesize log_enable;
@synthesize to_email;

- (void)setCredentialValues: (NSString *)credentials {
    username = [credentials valueForKey: @"username"];
    password = [credentials valueForKey: @"password"];
    domain = [credentials valueForKey: @"domain"];
    address = [credentials valueForKey: @"address"];
    proxy = [credentials valueForKey: @"proxy"];
    turnServer0 = [credentials valueForKey: @"turnServer0"];
    turnDomain0 = [credentials valueForKey: @"turnDomain0"];
    turnUsername0 = [credentials valueForKey: @"turnUsername0"];
    turnPassword0 = [credentials valueForKey: @"turnPassword0"];
    turnServer1 = [credentials valueForKey: @"turnServer1"];
    turnDomain1 = [credentials valueForKey: @"turnDomain1"];
    turnUsername1 = [credentials valueForKey: @"turnUsername1"];
    turnPassword1 = [credentials valueForKey: @"turnPassword1"];
    
    if (![[address substringToIndex: 3] isEqualToString: @"sip"]) {
        address = [@"sip:" stringByAppendingString: address];
    }
}

- (void)setCallSettingValues: (NSString *)settings {
    onlyAudioCall = [[settings valueForKey:@"only_audio"] intValue];
    download_bandwidth = [[settings valueForKey: @"download_bandwidth"] intValue];
    upload_bandwidth = [[settings valueForKey: @"upload_bandwidth"] intValue];
    framerate = [[settings valueForKey: @"framerate"] intValue];
    transport_mode = [settings valueForKey:@"transport_mode"];
    encryption_mode = [settings valueForKey:@"encryption_mode"];
    video_size = [settings valueForKey:@"video_size"];
}

- (void)setGUISettings: (NSString *)guiSettings {
    language = [guiSettings valueForKey:@"language"];
    main_color = [guiSettings valueForKey:@"main_color"];
    secondary_color = [guiSettings valueForKey:@"secondary_color"];
    font_size = [[guiSettings valueForKey:@"font_size"] intValue];
    font_color = [guiSettings valueForKey:@"font_color"];
    display_topview_mode = [guiSettings valueForKey:@"display_topview_mode"];
    consultant_name = [guiSettings valueForKey:@"consultant_name"];
    call_recording_notification_visible = [guiSettings valueForKey:@"call_recording_notification_visible"];
    chat_mode = [guiSettings valueForKey:@"chat_mode"];
    zoom_mode = [guiSettings valueForKey:@"zoom_mode"];
    log_enable = [[guiSettings valueForKey:@"log_mode"] boolValue];
    [self checkDefaultValues];
}

- (void)setExtraSettings: (NSString *)extraSettings {
    videocall_version = [extraSettings valueForKey:@"videocall_version"];
    enduser_version = [extraSettings valueForKey:@"enduser_version"];
    consultation_id = [extraSettings valueForKey:@"consultation_id"];
    to_email = [extraSettings valueForKey:@"extra_toemail"];
}

- (void)checkDefaultValues {
    if (display_button_time <= 0) {
        display_button_time = 3;
    }
    if ([StringUtil isEmpty:language]) {
        language = @"es";
    }
    if ([StringUtil isEmpty:main_color]) {
        main_color = @"0a79c7";
    }
    if ([StringUtil isEmpty:secondary_color]) {
        secondary_color = @"70c8e1";
    }
    if ([StringUtil isEmpty:display_topview_mode]) {
        display_topview_mode = @"atScreenTouch";
    }
    if ([StringUtil isEmpty:font_color]) {
        font_color = @"ffffff";
    }
    if ([StringUtil isEmpty:consultant_name]) {
        consultant_name = @"John Doe";
    }
    if (font_size <= 0) {
        font_size = 17;
    }
    if ([StringUtil isEmpty:call_recording_notification_visible]) {
        call_recording_notification_visible = @"no";
    }
}

@end
